lib/tests/streamlit/elements/data_editor_test.py

```diff
@@ -938,3 +938,16 @@ def test_shows_cached_widget_replay_warning(self):
         el = self.get_delta_from_queue(-2).new_element.exception
         assert el.type == "CachedWidgetWarning"
         assert el.is_warning
+
+    def test_preserves_int_column_labels_when_no_edit(self) -> None:
+        """Test that integer column labels are preserved in the round-trip through the editor."""
+        # Create a DataFrame with integer column names
+        df = pd.DataFrame([[1, 2]], columns=[0, 1])
+
+        # Round-trip through the editor without renaming any columns
+        edited = st.data_editor(df, key="no_edit_int_cols")
+
+        # Expect the same integer column names back
+        assert list(edited.columns) == [0, 1]
+        assert all(isinstance(col, int) for col in edited.columns)
```

Comment:
I would get rid of the docstring + comments here since I don't think they add much.

Also, I would remove the `_when_no_edit` at the end of the test name as I'm fairly sure we don't have to worry about the column name editing case + things will still work if you edit the values in the column (see my other comment)